Library/Homebrew/cask/lib/hbc/utils/ls_quarantine/extended_attributes.rb

```diff
@@ -0,0 +1,67 @@
+require "open3"
+
+# class for reading and writing extended attributes
+module LSQuarantine
+  class ExtendedAttributes
+    def initialize(file, follow_symlinks: false)
+      @file = File.new(file)
+      @follow_symlinks = follow_symlinks
+    end
+
+    def list
+      out, status = command
+
+      return nil unless status.success?
+      return [] if out.empty?
+
+      out.chomp.split(/\n/)
+    end
+
+    def to_h
+      lst = list
+
+      return nil if lst.nil?
+
+      list.each_with_object({}) do |attribute, hash|
+        hash[attribute] = get(attribute)
+      end
+    end
+
+    def set(name, content)
+      command("-w", name, content)[1].success?
+    end
+
+    def get(name)
+      out, status = command("-x", "-p", name)
+
+      return nil unless status.success?
+
+      # `xattr -x` outputs hex-string with line breaks and spaces
+      [out.delete("\n\s")].pack("H*")
+    end
+
+    def remove(name)
+      command("-d", name)
+      !(list || []).include?(name)
+    end
+
+    def clear
+      command("-c")[1].success?
+    end
+
+    private
+
+    XATTR_EXECUTABLE = Open3.capture2("which", "xattr")[0].chomp
+
+    def command(*args)
+      raise StandardError, "xattr not installed." if XATTR_EXECUTABLE.empty?
```

Comment:
Not sure if this or `which xattr` is needed as we use `/usr/bin/xattr` for [add_altname_metadata](https://github.com/Homebrew/brew/blob/f3ec40d696b421e89f462d32f6f30a81f60d3e73/Library/Homebrew/cask/lib/hbc/artifact/relocated.rb#L56-L73)